From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Wed, 26 Oct 2016 14:23:35 -0400
Subject: link with --as-needed

---
 configure/CONFIG_SITE | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure/CONFIG_SITE b/configure/CONFIG_SITE
index 0c9da2e..9352c76 100644
--- a/configure/CONFIG_SITE
+++ b/configure/CONFIG_SITE
@@ -25,3 +25,5 @@
 #INSTALL_LOCATION=<fullpathname>
 
 #STATIC_BUILD=NO
+
+USR_LDFLAGS += -Wl,--as-needed
