From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Wed, 26 Oct 2016 14:23:35 -0400
Subject: link with --as-needed

---
 configure/CONFIG_SITE | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure/CONFIG_SITE b/configure/CONFIG_SITE
index 5ee9934..759779b 100644
--- a/configure/CONFIG_SITE
+++ b/configure/CONFIG_SITE
@@ -34,3 +34,5 @@ USR_CPPFLAGS += -DUSE_TYPED_RSET
 
 # Set to YES to link executable (mostly) statically
 #STATIC_BUILD=NO
+
+USR_LDFLAGS += -Wl,--as-needed
