From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Wed, 26 Oct 2016 14:28:53 -0400
Subject: mrf.ko advertise pci id table

---
 mrmShared/linux/uio_mrf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mrmShared/linux/uio_mrf.c b/mrmShared/linux/uio_mrf.c
index eddbb50..d6f129b 100644
--- a/mrmShared/linux/uio_mrf.c
+++ b/mrmShared/linux/uio_mrf.c
@@ -704,6 +704,7 @@ static struct pci_device_id mrf_pci_ids[] = {
     },
     { 0, }
 };
+MODULE_DEVICE_TABLE(pci, mrf_pci_ids);
 
 /************************** Module boilerplate ****************************/
 
