From: Michael Davidsaver <mdavidsaver@gmail.com>
Date: Wed, 9 Nov 2016 17:19:11 -0500
Subject: remove verbose irq print

---
 mrmShared/linux/uio_mrf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mrmShared/linux/uio_mrf.c b/mrmShared/linux/uio_mrf.c
index d6f129b..ef1df76 100644
--- a/mrmShared/linux/uio_mrf.c
+++ b/mrmShared/linux/uio_mrf.c
@@ -287,10 +287,10 @@ mrf_handler_plx(int irq, struct uio_info *info)
         flags |= IRQ_Enable_ALL;
 
         if((flags & val)==0) {
-            dev_info(&dev->dev, "reject %08x %08x\n", (unsigned)flags, (unsigned)val);
+            dev_dbg(&dev->dev, "reject %08x %08x\n", (unsigned)flags, (unsigned)val);
             return IRQ_NONE; /* not our interrupt */
         } else
-            dev_info(&dev->dev, "accept %08x %08x\n", (unsigned)flags, (unsigned)val);
+            dev_dbg(&dev->dev, "accept %08x %08x\n", (unsigned)flags, (unsigned)val);
 
         if(!priv->usemie) {
             // Disable interrupts on FPGA
